cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedStmt.qll

```diff
@@ -711,3 +711,81 @@ class TranslatedSwitchStmt extends TranslatedStmt {
     child = getBody() and result = getParent().getChildSuccessor(this)
   }
 }
+
+class TranslatedAsmStmt extends TranslatedStmt {
+  override AsmStmt stmt;
+
+  override TranslatedElement getChild(int id) {
+    none()
+  }
+
+  override Instruction getFirstInstruction() {
+    if exists(stmt.getChild(0))
+    then result = getInstruction(AsmInputTag(0))
+    else result = getInstruction(AsmTag())
+  }
+
+  override predicate hasInstruction(Opcode opcode, InstructionTag tag,
+      Type resultType, boolean isGLValue) {
+    tag = AsmTag() and
+    opcode instanceof Opcode::InlineAsm and
+    resultType instanceof UnknownType and
+    isGLValue = false
+    or
+    exists(int index, VariableAccess va |
```

Comment:
Is the idea here that by consuming the addresses of the variables, the variables will be considered as escaped, and the read and write of "all aliased memory" will model the potential side effects correctly?

Reply:
Yes, exactly